ledger/src/bank_forks_utils.rs

```diff
@@ -199,11 +206,9 @@ fn bank_forks_from_snapshot(
         process::exit(1);
     }
 
-    let (deserialized_bank, full_snapshot_archive_info, incremental_snapshot_archive_info) =
-        snapshot_utils::bank_from_latest_snapshot_archives(
+    let file_ret = if snapshot_config.snapshot_from == SnapshotFrom::File {
```

Comment:
I think these blocks can be cleaned up - it's odd to have several maps from `Ok(good_ret) => Some(good_ret)` and makes it slightly more confusing as what's happening here, i.e. if SnapshotFrom::File try loading from files, fallback to loading archives, expect we have a valid bank at the end, regardless of path.

Reply:
Here the flow is:
if the snapshot_config.snapshot_from is not File, then the first stage result is none.
otherwise, try the file route, which can return valid result or none.

For both the above, if the result is none, do the archive way.

Not sure how I can express this better in Rust.

Reply:
Could do something like:
`(snapshot_config.snapshot_from == SnapshotFrom::File).map(|| ...).map_err(|err| warn!(...) err).ok().unwrap_or(|| ...);`

`map_err` is kinda a hack here though, we'd really want `inspect_err` which isn't stable yet, unfortunately